trieste/models/model_interfaces.py

```diff
@@ -300,6 +314,9 @@ def update(self, dataset: Dataset):
     def predict(self, query_points: QueryPoints) -> Tuple[ObserverEvaluations, TensorType]:
         return self.model.predict_y(query_points)
 
+    def predict_full_cov(self, query_points: QueryPoints) -> Tuple[ObserverEvaluations, TensorType]:
+        return self.model.predict_y(query_points, full_cov=True)
```

Comment:
mmmmm not sure this is the right call! I would remove those lines so that predict_full_cov always returns samples from the latent models.

Reply:
ok. Presumably we should then also change `predict` so it matches `predict_full_cov`?

Reply:
no no no! the other predict has to be y

Reply:
ok. so I suggest we have separate methods for whether you've included the likelihood or not:

`predict_f` (or `predict_latent`)
`predict_y` (or `predict_observations`)
`predict_f_samples` (or `predict_latent_samples`)
`predict_y_samples` (or `predict_observations_samples`, not needed yet)

Given that's a lot of change, how about we rename `predict_full_cov` to `predict_latent_full_cov` for clarity?

Reply:
looking back at this, I still don't follow why we're doing f for one and y for the other (never mind that `sample` is also sampling y atm)